// @crave/farmblocks-theme/lib/colors.js
"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = exports.TANGERINE_2 = exports.TANGERINE = exports.WHITE_56 = exports.WHITE_32 = exports.GREY_32 = exports.GREY_16 = exports.GREY_08 = exports.EGGPLANT = exports.GRAPE = exports.TOMATO = exports.STRAWBERRY = exports.COFFEE = exports.CARAMEL = exports.CARROT = exports.PUMPKIN = exports.BARLEY = exports.CORN = exports.AVOCADO = exports.LETTUCE = exports.BLUE_CORN = exports.INDIGO_MILK_CAP = exports.RED_ORANGE = exports.DEMERARA_SUGAR = exports.SUGAR = exports.OYSTER = exports.CARBON = void 0;
var CARBON = "#2f313a";
exports.CARBON = CARBON;
var OYSTER = "#59636f";
exports.OYSTER = OYSTER;
var SUGAR = "#f6f8f9";
exports.SUGAR = SUGAR;
var DEMERARA_SUGAR = "#e9ebec";
exports.DEMERARA_SUGAR = DEMERARA_SUGAR;
var RED_ORANGE = "#ff4411";
exports.RED_ORANGE = RED_ORANGE;
var INDIGO_MILK_CAP = "#3498db";
exports.INDIGO_MILK_CAP = INDIGO_MILK_CAP;
var BLUE_CORN = "#2980b9";
exports.BLUE_CORN = BLUE_CORN;
var LETTUCE = "#2ecc71";
exports.LETTUCE = LETTUCE;
var AVOCADO = "#27ae60";
exports.AVOCADO = AVOCADO;
var CORN = "#f1c618";
exports.CORN = CORN;
var BARLEY = "#ccaa00";
exports.BARLEY = BARLEY;
var PUMPKIN = "#e67e22";
exports.PUMPKIN = PUMPKIN;
var CARROT = "#d35400";
exports.CARROT = CARROT;
var CARAMEL = "#9e5500";
exports.CARAMEL = CARAMEL;
var COFFEE = "#774411";
exports.COFFEE = COFFEE;
var STRAWBERRY = "#dd3311";
exports.STRAWBERRY = STRAWBERRY;
var TOMATO = "#bc2b0e";
exports.TOMATO = TOMATO;
var GRAPE = "#9b59b6";
exports.GRAPE = GRAPE;
var EGGPLANT = "#8e44ad";
exports.EGGPLANT = EGGPLANT;
var GREY_08 = "rgba(0, 0, 0, 0.08)";
exports.GREY_08 = GREY_08;
var GREY_16 = "rgba(0, 0, 0, 0.16)";
exports.GREY_16 = GREY_16;
var GREY_32 = "rgba(0, 0, 0, 0.32)";
exports.GREY_32 = GREY_32;
var WHITE_32 = "rgba(255, 255, 255, 0.32)";
exports.WHITE_32 = WHITE_32;
var WHITE_56 = "rgba(255, 255, 255, 0.56)";
exports.WHITE_56 = WHITE_56;
var TANGERINE = "#fd6a08";
exports.TANGERINE = TANGERINE;
var TANGERINE_2 = "#E36009";
exports.TANGERINE_2 = TANGERINE_2;
var _default = {
  CARBON: CARBON,
  OYSTER: OYSTER,
  SUGAR: SUGAR,
  DEMERARA_SUGAR: DEMERARA_SUGAR,
  RED_ORANGE: RED_ORANGE,
  INDIGO_MILK_CAP: INDIGO_MILK_CAP,
  BLUE_CORN: BLUE_CORN,
  LETTUCE: LETTUCE,
  AVOCADO: AVOCADO,
  CORN: CORN,
  BARLEY: BARLEY,
  PUMPKIN: PUMPKIN,
  CARROT: CARROT,
  CARAMEL: CARAMEL,
  COFFEE: COFFEE,
  STRAWBERRY: STRAWBERRY,
  TOMATO: TOMATO,
  GRAPE: GRAPE,
  EGGPLANT: EGGPLANT,
  GREY_08: GREY_08,
  GREY_16: GREY_16,
  GREY_32: GREY_32,
  WHITE_32: WHITE_32,
  WHITE_56: WHITE_56,
  TANGERINE: TANGERINE,
  TANGERINE_2: TANGERINE_2
};
exports["default"] = _default;
